src/storage.rs

```diff
@@ -21,8 +21,9 @@ const LOCK_FILE: &str = "pearl.lock";
 ///
 #[derive(Debug)]
 pub struct Storage<K> {
+    initialized: Option<()>,
     config: Config,
-    active_blob: Box<Option<Blob<K>>>,
+    active_blob: Option<Box<Blob<K>>>,
```

Comment:
This field is accessed from multiple threads thus it is required to use synchronization or memory barrier. In current shape it is not a pointer, so all access should happen inside critical section. If you can make it a pointer size type, then `volatile` should be enough to prevent torn read.

Reply:
need to think about good rust synchronization mechanism here. Mutex or RwLock (https://doc.rust-lang.org/std/sync/struct.RwLock.html) as we can read independently (there is a syscall pread in nix which can read with offset without need to seek operation https://docs.rs/nix/0.13.0/nix/sys/uio/fn.pread.html)

Reply:
https://docs.rs/crossbeam/*/crossbeam/atomic/struct.AtomicCell.html

Reply:
Syncronization and concurrency will be included in one of the followin issues

Reply:
it will be in #8